### PR TITLE
Vectorize generation sampling and report tok/s from generate_text

### PR DESCRIPTION
**Pull request by workofart**
## Summary
- `generate()` sampled each completion row in a Python loop with per-row host syncs every step (`int(argmax)`, `to_scalar` on the sampled id, another on its logprob). The sampling contract moves into `_sample_next_tokens` — greedy vs sampled decoding, top-k masking, EOS padding for finished rows, logprobs — with the math batched across rows, so each decode step does one device-to-host transfer for the whole batch. Behavior is unchanged: all existing generation tests pass unmodified.
- New `compute_logprobs` flag on `generate()` lets callers that never read logprobs skip the per-step log-softmax work. `generate_text` passes `False` since text decoding only needs the tokens; the GRPO path keeps the default `True`.
- `generate_text` now prints the decoded prompt, a labeled completion, and a one-line wall-time / tok/s summary, giving generation runs a stable throughput number independent of tqdm's per-iter overhead.

## Evidence
From the experimental-branch inference notes (GPT-2 124M, MLX): per-step logprob math (max/sub/exp/sum/log/gather) cost ~15% of decode wall time at 250+ tok/s — skipping it via `compute_logprobs=False` moved measured generation from ~225 to ~260 tok/s. That measurement was on the KV-cached decode path, where sampling is a large fraction of step cost; on this branch's eager path the forward pass dominates, so the immediate win is smaller — the flag and the batched sampling mainly remove per-row host syncs that scale with `num_generations`.

## Test plan
- `test_generate_can_skip_logprob_math` pins that `compute_logprobs=False` returns 0.0 logprobs and identical tokens.
- `test_generate_text_prints_prompt_completion_and_token_counts` pins the prompt/completion labels and token-count summary.
- Existing sampling tests (logprob-from-sampling-distribution, batched parallel completions, EOS stop) pass unchanged on both MLX and numpy backends.
- Full suite: 561 passed, 11 skipped.

### CODE REVIEW

File: autograd/text/utils.py [hunk 1]
```diff
@@ -143,58 +147,121 @@ def generate(
...
+    next_token_logits: Array,
+    active: list[bool],
+    temperature: float,
+    top_k: Optional[int],
+    eos_token_id: int,
+    compute_logprobs: bool,
+) -> tuple[Array, Optional[Array]]:
+    """Sample one token per row from next-token logits.
+
+    This owns the sampling contract for the generation loop: greedy vs sampled
```
Comment (workofart) [inline autograd/text/utils.py:200]: previously the different mechanisms for sampling lives in many different places. Now, we're consolidating them here.
